src/common/components/repository-input/index.js

```diff
@@ -0,0 +1,70 @@
+import 'isomorphic-fetch';
+
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+import {
+  changeRepositoryInput,
+  verifyGitHubRepository
+} from '../../actions/repository-input';
+
+export class RepositoryInput extends Component {
+  constructor(props) {
+    super(props);
+  }
+
+  getStatusMessage() {
```

Comment:
I think this is good example to discuss the limits of presentation component logic. I think this amount of switching is okay, but if it became any more complex it would be better to move it into the store.

From a testing perspective, it's nice to pass props like "message" and "status" to presentational components and test whether they are used. But I'm borderline in this instance.